### PR TITLE
Unit Testing, Progress Report, Class Quizzes, Quiz Preview & Results, Travis, codecov

### LINKED ISSUE #12: Create View Progress Report Site
Static Site to display personalized progress reports for individual students. Need to come up with a scoring matrix.

### LINKED ISSUE #14: Create View Others' Progress Report Feature
Allow students/educators to see the others' (only if they are in the same class) progress report.

### LINKED ISSUE #28: Create Update Participants List Feature
Allow educators to manually include students if necessary.

### LINKED ISSUE #60: Create Quizzes Page for Class
To be able to view quizzes from within the classes. Can contain quizzes created by the educator of the class or the application's own tailored quizzes.

### LINKED ISSUE #65: Styling of Quiz Preview / Results
- `previewquiz.html`
- `result.html`

Include #26 Edit Question, #27 Remove Question in `previewquiz.html`.

### PR DESCRIPTION
**Pull request by hughjazzman**
closes #28, closes #60, closes #65 

#12 #14 not broken, but will have to 
- convert `datetime` to use in JSON for overall proficiency
- get topical data as a list of 2 lists with first being topics, 2nd being proficiency level

closes #14 as the routing for other users' reports works

#34 covered greatly, still more test cases to be written for Milestone 3.

Travis and codecov also added in this PR.

### CODE REVIEW

File: app/forms.py [hunk 1]
```diff
@@ -1,55 +1,49 @@
...
-    #        raise ValidationError('Please use a different username.')
-
     def validate_email(self, email):
         user = User.query.filter_by(email=email.data).first()
         if user is not None:
             raise ValidationError('Please use a different email address.')
 
-class ContactForm(FlaskForm):
+class ContactForm(FlaskForm, EmailFormMixin):
     name = StringField('Name', validators=[DataRequired()])
```
Comment (R-Ramana) [inline app/forms.py:34]: If we are going to use Mixin, we could also use the StringFormMixin for the string fields like name as well right?

Comment (hughjazzman) [reply app/forms.py:34]: I suppose. Not important for now. Would prevent the use of `form.label` if wanted.

Comment (R-Ramana) [reply app/forms.py:34]: Yea, I approved it. Maybe can think about it in Milestone 3 xD

**Comment by hughjazzman** [comment]
Testing was documented in [Developer Guide](https://github.com/Orbital-Knewbie/Knewbie/blob/master/docs/DeveloperGuide.md#test).

**Pull request merged**